### PR TITLE
[LM-9213] Customisable UI for topic feed

### PR DESCRIPTION
**Pull request by TyagiLikeMinds**

### CODE REVIEW

File: lib/src/models/topic_view_model.dart [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import 'package:likeminds_feed/likeminds_feed.dart';
+
+class TopicViewModel {
```
Comment (ishaan1607) [inline lib/src/models/topic_view_model.dart:3]: class name should be `TopicUI`

File: lib/src/models/topic_view_model.dart [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  String name;
+  String id;
+  bool isEnabled;
+
+  TopicViewModel(
+      {required this.name, required this.id, required this.isEnabled});
```
Comment (ishaan1607) [inline lib/src/models/topic_view_model.dart:9]: we should builder pattern to these classes

File: lib/src/views/feed_page.dart
Comment (ishaan1607) [inline lib/src/views/feed_page.dart:1]: why empty file?

File: lib/src/views/feedroom_page.dart
Comment (ishaan1607) [inline lib/src/views/feedroom_page.dart:1]: why empty file?

File: lib/src/views/new_post_page.dart
Comment (ishaan1607) [inline lib/src/views/new_post_page.dart:1]: why empty file?

File: lib/src/widgets/reply/reply_menu.dart
Comment (ishaan1607) [inline lib/src/widgets/reply/reply_menu.dart:1]: why empty file?

File: lib/src/widgets/reply/reply_tile.dart
Comment (ishaan1607) [inline lib/src/widgets/reply/reply_tile.dart:1]: why empty file?

File: lib/src/widgets/topic/topic_chip.dart [hunk 3]
```diff
@@ -0,0 +1,131 @@
+import 'package:flutter/material.dart';
+import 'package:likeminds_feed_ui_fl/likeminds_feed_ui_fl.dart';
+import 'package:likeminds_feed_ui_fl/src/utils/theme.dart';
+
+// defines icon placement
+// LMIconPlacement.start places the icon before the text
+// LMIconPlacement.end places the icon after the text
+enum LMIconPlacement { start, end }
```
Comment (ishaan1607) [inline lib/src/widgets/topic/topic_chip.dart:8]: this should be in separate file.

File: lib/src/widgets/topic/topic_feed_bar.dart [hunk 4]
```diff
@@ -0,0 +1,200 @@
+import 'package:flutter/material.dart';
+import 'package:likeminds_feed_ui_fl/likeminds_feed_ui_fl.dart';
+import 'package:likeminds_feed_ui_fl/src/utils/theme.dart';
+
+// This widget is used to display a topic feed bar
+// A [TopicFeedBar] displays a list of selected topics
+// The [TopicFeedBar] can be customized by passing in the required parameters
+class TopicFeedBar extends StatelessWidget {
```
Comment (ishaan1607) [inline lib/src/widgets/topic/topic_feed_bar.dart:8]: it should be `LMTopicFeedBar`

File: lib/src/widgets/topic/topic_feed_grid.dart [hunk 5]
```diff
@@ -0,0 +1,163 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_layout_grid/flutter_layout_grid.dart';
+import 'package:likeminds_feed_ui_fl/likeminds_feed_ui_fl.dart';
+import 'package:likeminds_feed_ui_fl/src/utils/theme.dart';
+
+// This widget is used to display a topic grid
+// A [TopicFeedGrid] displays a grid of selected topics
+// The [TopicFeedGrid] can be customized by passing in the required parameters
+class TopicFeedGrid extends StatelessWidget {
```
Comment (ishaan1607) [inline lib/src/widgets/topic/topic_feed_grid.dart:9]: it should be `LMTopicFeedGrid`

File: lib/src/widgets/topic/topic_tile.dart [hunk 6]
```diff
@@ -0,0 +1,64 @@
+import 'package:flutter/material.dart';
+import 'package:likeminds_feed_ui_fl/likeminds_feed_ui_fl.dart';
+import 'package:likeminds_feed_ui_fl/src/utils/theme.dart';
+
+// Displays a tile for a topic
+// The tile can be customized by passing in the required parameters
+// Tile has a text and an icon
+// The tile can be tapped to perform an action
+// Icon is displayed only if the tile is selected
+class TopicTile extends StatelessWidget {
```
Comment (ishaan1607) [inline lib/src/widgets/topic/topic_tile.dart:10]: it should be `LMTopicTile`

**Review by ishaan1607 — CHANGES_REQUESTED** [review]

File: lib/src/views/new_post_page.dart
Comment (TyagiLikeMinds) [reply lib/src/views/new_post_page.dart:1]: this is a deleted file

File: lib/src/widgets/reply/reply_menu.dart
Comment (TyagiLikeMinds) [reply lib/src/widgets/reply/reply_menu.dart:1]: deleted file

File: lib/src/widgets/topic/topic_feed_bar.dart (hunk 4 above)
Comment (TyagiLikeMinds) [reply lib/src/widgets/topic/topic_feed_bar.dart:8]: okay

File: lib/src/models/topic_view_model.dart (hunk 2 above)
Comment (TyagiLikeMinds) [reply lib/src/models/topic_view_model.dart]: added

File: lib/src/widgets/topic/topic_tile.dart (hunk 6 above)
Comment (TyagiLikeMinds) [reply lib/src/widgets/topic/topic_tile.dart]: changed

File: lib/src/widgets/topic/topic_feed_grid.dart (hunk 5 above)
Comment (TyagiLikeMinds) [reply lib/src/widgets/topic/topic_feed_grid.dart]: changed

File: lib/src/models/topic_view_model.dart (hunk 1 above)
Comment (TyagiLikeMinds) [reply lib/src/models/topic_view_model.dart]: changed

File: lib/src/widgets/reply/reply_tile.dart
Comment (TyagiLikeMinds) [reply lib/src/widgets/reply/reply_tile.dart:1]: this is a deleted file

**Pull request merged**